packages/plugins/typescript/resolvers/tests/federation.spec.ts

```diff
@@ -45,16 +45,45 @@ describe('TypeScript Resolvers Plugin + Apollo Federation', () => {
       schema: federatedSchema,
       config: {
         federation: true,
+        mappers: {
+          User: './my-models#UserModel',
+        },
       },
     });
 
     // User should have it
     expect(content).toBeSimilarStringTo(`
-      __resolveReference?: ReferenceResolver<Maybe<ResolversTypes['User']>, { __typename: 'User' } & GraphQLRecursivePick<ParentType, {"id":true}>, ContextType>;
+      __resolveReference?: ReferenceResolver<Maybe<ParentType>, { __typename: 'User' } & GraphQLRecursivePick<ParentType, {"id":true}>, ContextType>;
     `);
-    // Foo shouldn't because it doesn't have @key
+    // Book shouldn't because it doesn't have @key
     expect(content).not.toBeSimilarStringTo(`
-      __resolveReference?: ReferenceResolver<Maybe<ResolversTypes['Book']>, { __typename: 'Book' } & GraphQLRecursivePick<ParentType, {"id":true}>, ContextType>;
+      __resolveReference?: ReferenceResolver<Maybe<ParentType>, { __typename: 'Book' } & GraphQLRecursivePick<ParentType, {"id":true}>, ContextType>;
+    `);
+  });
+
+  it('should not add __resolveReference to objects that do not have mapped types in config.mappers', async () => {
```

Comment:
This is not true. The `__resolveReference` should be available for all types with `@key` that are local to the service.